reactive/kubernetes_worker.py

```diff
@@ -319,11 +308,37 @@ def join_or_update_cohorts():
     kube_control = endpoint_from_flag('kube-control.cohort_keys.available')
     cohort_keys = kube_control.cohort_keys
     for snapname in cohort_snaps:
-        hookenv.status_set('maintenance', 'Joining snap cohort.')
+        hookenv.status_set('maintenance', 'Joining cohort for {}.'.format(snapname))
         cohort_key = cohort_keys[snapname]
-        snap.join_cohort_snapshot(snapname, cohort_key)
-    hookenv.log('{} has joined the snap cohort'.format(hookenv.local_unit()))
+        for delay in (5, 30, 60):
+            try:
+                snap.join_cohort_snapshot(snapname, cohort_key)
+                hookenv.log('Joined cohort for {}'.format(snapname))
+                break
+            except subprocess.CalledProcessError:
+                hookenv.log('Error joining cohort for {}'.format(snapname),
+                            level=hookenv.ERROR)
+                hookenv.status_set('maintenance',
+                                   'Error joining cohort for {} (see logs), '
+                                   'will retry.'.format(snapname))
+                time.sleep(delay)
```

Comment:
It seems to me like this would cause us to hammer the snap store even more. I would think that we would want to remove the snap refresh on update-status, instead firing maybe every 60 minutes, and then have this retry logic for during an actual upgrade.